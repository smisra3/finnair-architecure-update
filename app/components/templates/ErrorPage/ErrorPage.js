// @flow
import { Component } from 'react';
import get from 'lodash/get';
import Layout from '../Layout';
import enhance from '../../../lib/dynamicStore';
import initialActions from './ErrorPage.actions';
import HeadTag from '../../atoms/HeadTag';
import saga from './ErrorPage.saga';
import reducer from './ErrorPage.reducer';

class ErrorPage extends Component<any> {
  static getInitialProps({ res }: any) {
    if (res && res.redirect) {
      res.status(404);
    }
    return {};
  }

  render() {
    const { errorData } = this.props;
    const errorMessage = get(errorData, ['message']);
    return (
      <Layout title="error" className="row" id="content-wrapper" tabindex="-1">
        <HeadTag description="error page" title="error page" />
        <section id="notFoundContent" style={{ width: '100%' }}>
          {errorMessage}
        </section>
      </Layout>
    );
  }
}

/* istanbul ignore next */
const mapStateToProps = state => ({
  errorData: get(state, ['errorPage', 'errorPageData']),
});

const mapDispatchToProps = dispatch => ({});

export default enhance(ErrorPage, {
  mapStateToProps,
  mapDispatchToProps,
  saga,
  reducer,
  key: 'errorPage',
  initialActions,
});

export { ErrorPage as ErrorPageDisconnected };
