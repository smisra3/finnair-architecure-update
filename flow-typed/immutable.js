// @flow
// For global react
declare var React: $Exports<'react'>;
declare module 'immutable' {
  declare module.exports: any;
}
