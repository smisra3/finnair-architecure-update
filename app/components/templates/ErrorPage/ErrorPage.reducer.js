// @flow
import { updateState } from '../../../utils/utils';
import { LOAD_ERROR_PAGE_SUCCESS, LOAD_ERROR_PAGE_FAILURE } from './ErrorPage.constants';

export const setErrorPageData = (state, data: any) => updateState(state, { errorPageData: data });

export const layout = (state: Object = {}, action: any = 'default') => {
  switch (action.type) {
    case LOAD_ERROR_PAGE_SUCCESS:
      return setErrorPageData(state, action.data);
    case LOAD_ERROR_PAGE_FAILURE:
      return setErrorPageData(state, 'Error');
    default:
      return state;
  }
};

export default layout;
