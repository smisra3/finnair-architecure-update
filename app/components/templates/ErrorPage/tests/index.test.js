import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import { ErrorPageDisconnected as ErrorPage } from '../ErrorPage';

const props = {
  errorData: fromJS({
    message: 'Test Message',
  }),
};

describe('ErrorPage Component', () => {
  test('should render correctly', () => {
    const errorPage = shallow(<ErrorPage {...props} />);
    expect(errorPage).toMatchSnapshot();
  });
});
