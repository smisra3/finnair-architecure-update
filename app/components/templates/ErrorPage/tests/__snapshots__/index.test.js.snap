// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage Component should render correctly 1`] = `
<Connect(Layout)
  className="row"
  id="content-wrapper"
  tabindex="-1"
  title="error"
>
  <HeadTag
    description="error page"
    title="error page"
  />
  <section
    id="notFoundContent"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    Test Message
  </section>
</Connect(Layout)>
`;
